### PR TITLE
py-git2: New Package

### PR DESCRIPTION
**Pull request by citibeth**

### CODE REVIEW

File: var/spack/repos/builtin/packages/py-git2/package.py [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+##############################################################################
+from spack import *
+
+
+class PyGit2(Package):
+    """Pygit2 is a set of Python bindings to the libgit2 shared library,
+    libgit2 implements the core of Git.
+    """
+
+    # FIXME: Add a proper url for your package's homepage here.
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-git2/package.py]: Remove this FIXME

File: var/spack/repos/builtin/packages/py-git2/package.py [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+    homepage = "http://www.pygit2.org/"
+    url      = "https://pypi.python.org/packages/aa/56/84dcce942a48d4b7b970cfb7a779b8db1d904e5ec5f71e7a67a63a23a4e2/pygit2-0.24.1.tar.gz"
+
+    version('0.24.1', 'dd98b6a9fded731e36ca5a40484c8545')
+
+    extends('python')
+    depends_on('libgit2@0.24.2')
+    depends_on('py-cffi', type=nolink)
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-git2/package.py]: Looking at the `setup.py`, this package also requires `py-setuptools` as a build dependency and `py-six` as a nolink dependency.

**Comment by citibeth** [comment]
@tgamblin This will be a prerequisite to the advanced features being thrown around in #2179 
(I'll make sure that Spack still works as much as possible even if `py-git2` has not been built.)

**Comment by tgamblin** [comment]
@citibeth ok.  FYI, though, Spack calls git manually and I've found that to be generally more reliable than going through a library.  I would be hesitant to add a dependency for this.

**Comment by citibeth** [comment]
@tgamblin  I have the same concerns.  But... given that the git version scheme is currently speculative and will need some experimentation to get right, I wanted to give myself the most powerful tool available, at least for the prototype.  I've done past git hacking projects both with command-line git and with `py-git2`.  Regardless of how this issue is ultimately addressed, `py-git2` seemed like a reasonable addition to Spack's package repo.

**Comment by citibeth** [comment]
What's the best practice?  I invented `CMakePackage`, but so far I'm only
motivated to use it in cases I want to run `spack setup`.

**Comment by adamjstewart** [comment]
I think the best practice will be to move as many packages as we can to CmakePackage and AutotoolsPackage, especially new packages. I don't want to force anyone to use the new scheme, but when you use `spack create` it does this by default. Personally, I plan on switching packages to the new scheme whenever I happen to need to edit them or add a new version. But I think for the most part the new packages simplify the package.py a lot.

**Comment by citibeth** [comment]
@adamjstewart Flake8 is complaining about the URL line.  I though it wouldn't complain, but has it started complaining recently?  What is the best practice to deal with this Travis error?

**Comment by adamjstewart** [comment]
You may have to rebase. There was a bug introduced when we switched from `run-flake-tests` to `spack flake8`, but that was fixed a while ago.

**Comment by tgamblin** [comment]
Only thing holding this up is flake8.

**Comment by adamjstewart** [comment]
Ping @citibeth. I think this just needs a rebase before it will pass flake8 tests.

**Comment by citibeth** [comment]
This PR now needs #2778 before it will work.  I'd be happy to either include it into #2778 now (if @adamjstewart is OK with that), or we wait until #2778 is merged.

**Pull request merged**